app/scripts/ui.js

```diff
@@ -26,18 +26,44 @@ import metaRPCClientFactory from './lib/metaRPCClientFactory';
 
 const container = document.getElementById('app-content');
 
-const WORKER_KEEP_ALIVE_INTERVAL = 1000;
+// Service Worker Keep Alive Message Constants
+const WORKER_KEEP_ALIVE_INTERVAL = 1_000;
 const WORKER_KEEP_ALIVE_MESSAGE = 'WORKER_KEEP_ALIVE_MESSAGE';
-
+const ACK_KEEP_ALIVE_WAIT_TIME = 60_000; // 1 minute
+const ACK_KEEP_ALIVE_MESSAGE = 'ACK_KEEP_ALIVE_MESSAGE';
+let lastMessageRecievedTimestamp = Date.now();
 /*
  * As long as UI is open it will keep sending messages to service worker
  * In service worker as this message is received
  * if service worker is inactive it is reactivated and script re-loaded
  * Time has been kept to 1000ms but can be reduced for even faster re-activation of service worker
  */
 if (isManifestV3) {
-  setInterval(() => {
+  const handle = setInterval(() => {
     browser.runtime.sendMessage({ name: WORKER_KEEP_ALIVE_MESSAGE });
+
+    const timeoutHandle = setTimeout(() => {
+      if (
+        Date.now() - lastMessageRecievedTimestamp >
+        ACK_KEEP_ALIVE_WAIT_TIME
+      ) {
+        clearInterval(handle);
+        displayCriticalError(
+          'somethingIsWrong',
+          new Error("Something's gone wrong. Try reloading the page."),
+        );
+      }
+    }, ACK_KEEP_ALIVE_WAIT_TIME);
+
+    // add listener to receive ACK_KEEP_ALIVE_MESSAGE
+    // eslint-disable-next-line no-undef
+    const channel = new window.BroadcastChannel('sw-messages');
+    channel.addEventListener('message', (event) => {
```

Comment:
This would create a new channel and new event listener every second... is that okay?

Reply:
Yes, you're right, I have changed the code to initialise the Channel outside the listener (in app-init) and did same for the UI too.